doublets-ffi/examples/rust/examples/doublets-context.rs

```diff
@@ -0,0 +1,56 @@
+use doublets::{data::LinksConstants, Link};
+use doublets_ffi::{
+    constants::Constants,
+    export::{doublets_create_log_handle, doublets_free_log_handle},
+    store::{create, doublets_constants_u64, doublets_create_united_store_u64},
+    FFICallbackContext,
+};
+use std::{
+    ffi::{c_char, c_void, CStr, CString},
+    fs,
+    ptr::{null, null_mut},
+};
+
+unsafe extern "C" fn callback(_: FFICallbackContext, ptr: *const c_char) {
+    print!("{}", CStr::from_ptr(ptr).to_str().unwrap());
+}
+
+extern "C" fn create_cb<F>(ctx: FFICallbackContext, before: Link<u64>, after: Link<u64>) -> u64
+where
+    F: FnMut(Link<u64>, Link<u64>),
+{
+    unsafe {
+        let &mut (store, ref mut handler) = &mut *(ctx as *mut (*mut c_void, F));
```

Comment:
There is no reason to match it as a reference, just deref the pointer
```suggestion
        let (store, ref mut handler) = *(ctx as *mut (*mut c_void, F));
```

Reply:
Oh, bro, sorry, it is outdated. Currently i use [let handler = &mut *(ctx as *mut F);](
https://github.com/linksplatform/doublets-rs/pull/10/commits/f4c7cb4f6b602a5e75d67b928334d3fa6b9f3c7c#diff-d0218369fe3480c7622aed6587436e4df2f45a36faa7aca61056745ba60ffe34R27)